From 4e8121191b77f54b3d2b5db305ae16ea492d0fe9 Mon Sep 17 00:00:00 2001
From: Evan Wang <xswang@marvell.com>
Date: Mon, 5 Jun 2017 18:50:50 +0800
Subject: [PATCH 33/45] usb: gadget: u3d: add specific compatible entry for
 Armada 3700

The SoC of Armada 3700 requires different PHY HW reset
settings, compared to other Marvell SoCs, so it is necessary
to change the common compatible string to a new one,
and then it can be distinguished in the driver.

The patch changed the compatible string from
"marvell,mvebu-u3d" to "marvell,armada3700-u3d", and
updated the binding document accordingly.

Change-Id: I2d3ec4ed8952d1048efcb9dc1a0cc12d204c157f
Signed-off-by: Evan Wang <xswang@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/40250
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Victor Gu <xigu@marvell.com>
---
 .../devicetree/bindings/usb/marvell-u3d.txt       | 15 +++++++++++++--
 arch/arm64/boot/dts/marvell/armada-37xx.dtsi      |  9 +++++++++
 drivers/usb/gadget/udc/mvebu_u3d.c                |  1 +
 3 files changed, 23 insertions(+), 2 deletions(-)

diff --git a/Documentation/devicetree/bindings/usb/marvell-u3d.txt b/Documentation/devicetree/bindings/usb/marvell-u3d.txt
index f7d33dfae105..45f5102e4000 100644
--- a/Documentation/devicetree/bindings/usb/marvell-u3d.txt
+++ b/Documentation/devicetree/bindings/usb/marvell-u3d.txt
@@ -3,7 +3,8 @@ Marvell USB3 device controller
 This controller is integrated in Armada 38x/3700/8K.
 
 Required properties:
-- compatible	: Must be "marvell,mvebu-u3d"
+- compatible	: Must be "marvell,mvebu-u3d" except Armada 3700,
+		  for Armada 3700 it is "marvell,armada3700-u3d".
 - reg		: Physical base address and size of the USB3 device registers map.
 - interrupts	: Should contain single irq line of USB3 device controller
 - clocks	: the clock associated to this controller
@@ -16,7 +17,7 @@ Optional properties :
 - phys		: the COMPHY associated to this controller
 - phy-names:	: the name of COMPHY indicated by above property of phys
 
-Example:
+Example for all SoCs, except Armada-3700:
 	u3d@520000 {
 		compatible = "marvell,mvebu-u3d";
 		reg = <0x520000 0x2000>;
@@ -29,4 +30,14 @@ Example:
 		status = "disabled";
 	};
 
+Example for Armada-3700:
+	u3d@50000 {
+		compatible = "marvell,armada3700-u3d";
+		/* 0: 0x50000: USB 3.0 Device port 0: DEV_INFO_REG(0:15 - version_id) */
+		reg = <0x50000 0x2000>;
+		interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&sgateclk 17>;
+		status = "disabled";
+	};
+
 
diff --git a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
index b2faf0d342e3..be03749c17c0 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
@@ -333,6 +333,15 @@
 				status = "disabled";
 			};
 
+			u3d@50000 {
+				compatible = "marvell,armada3700-u3d";
+				/* 0: 0x50000: USB 3.0 Device port 0: DEV_INFO_REG(0:15 - version_id) */
+				reg = <0x50000 0x2000>;
+				interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&sb_periph_clk 20>;
+				status = "disabled";
+			};
+
 			usb3: usb@58000 {
 				compatible = "marvell,armada3700-xhci",
 				"generic-xhci";
diff --git a/drivers/usb/gadget/udc/mvebu_u3d.c b/drivers/usb/gadget/udc/mvebu_u3d.c
index 7977168347de..544f6854e61d 100644
--- a/drivers/usb/gadget/udc/mvebu_u3d.c
+++ b/drivers/usb/gadget/udc/mvebu_u3d.c
@@ -2598,6 +2598,7 @@ static void mvc2_shutdown(struct platform_device *dev)
 
 static const struct of_device_id mv_usb3_dt_match[] = {
 	{.compatible = "marvell,mvebu-u3d"},
+	{.compatible = "marvell,armada3700-u3d"},
 	{},
 };
 
-- 
2.35.3

